tests/h/views/activity_test.py

```diff
@@ -147,6 +147,27 @@ def test_init_returns_404_when_feature_turned_off(self,
         with pytest.raises(httpexceptions.HTTPNotFound):
             activity.GroupSearchController(group, pyramid_request)
 
+    def test_search_redirects_if_slug_wrong(self,
+                                            controller,
+                                            group,
+                                            pyramid_request):
+        """
+        If the slug in the URL is wrong it should redirect to the right one.
+
+        For example /groups/<pubid>/foobar redirects to /groups/<pubid>/<slug>.
+
+        The other 'group_read' views on h.views.groups do this, this tests that
+        the ones in h.views.activity do as well.
+
+        """
+        pyramid_request.matchdict['slug'] = 'wrong'
+
+        with pytest.raises(httpexceptions.HTTPMovedPermanently) as exc:
+            controller.search()
+
+        assert exc.value.location == '/groups/{pubid}/{slug}'.format(
+            pubid=group.pubid, slug=group.slug)
```

Comment:
If all of the group_read views were in a single controller class, which is how I think we ought to refactor it later, then the controller's `__init__()` could do the redirecting of `GET` requests in one place and we could test it in one place as well